cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/batch/connector/ConnectorSource.java

```
@@ -74,4 +79,37 @@ public void prepareRun(BatchSourceContext context) {
     context.setInput(Input.ofDataset(datasetName, arguments));
   }
 
+  @Override
+  public void onRunFinish(boolean succeeded, BatchSourceContext context) {
+    super.onRunFinish(succeeded, context);
+    try {
+      cleanupDataset(succeeded, context);
+    } catch (Exception e) {
```

Should specific exception be caught here?

---

We would not like a pipeline to fail if cleanup was unsuccessful, so I think catching & ignoring all exception should be fine.

---

This can throw multiple exception such as IOException, DatasetInstantiationException etc.

The idea is to perform the cleanup as (best-effort).  I don't want any exception to get propagated and can cause failures. Let me know wdyt